modules/caddyhttp/templates/frontmatter.go

```diff
@@ -30,6 +30,9 @@ func extractFrontMatter(input string) (map[string]interface{}, string, error) {
 	}
 	firstLine := input[firstLineStart:firstLineEnd]
 
+	// ensure residue windows newline is removed
+	firstLine = strings.Trim(firstLine, "\r")
```

Comment:
Should we just be generous and elide all extra whitespace? `strings.TrimSpace`

Reply:
Then maybe in the for loop with the" if b == '\n'" switch to "if unicode.IsSpace(b)"
Then the strings.Trim will be redundant. 
Would require some benchmarks but could be faster all in all.

Reply:
@oganga `unicode.IsSpace(b)` would also return true for literal space characters ` ` which is not what we want...

Reply:
Ahe...well then it is not a good idea :)
I have no strong feeling for either version of trim.